oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/AggregationQueryEsDAO.java

```diff
@@ -96,7 +96,7 @@ public AggregationQueryEsDAO(ElasticSearchClient client) {
         SearchSourceBuilder sourceBuilder = SearchSourceBuilder.searchSource();
 
         BoolQueryBuilder boolQueryBuilder = QueryBuilders.boolQuery();
-        sourceBuilder.query(boolQueryBuilder);
+        sourceBuilder.query(QueryBuilders.boolQuery().filter(boolQueryBuilder));
 
         boolQueryBuilder.must().add(QueryBuilders.rangeQuery(Metrics.TIME_BUCKET).lte(endTB).gte(startTB));
         boolQueryBuilder.must().add(QueryBuilders.termQuery(EndpointInventory.SERVICE_ID, serviceId));
```

Comment:
I am not familiar with ES, but I am very familiar with Lucene.
Something likes TimeRange used `filter` is OK. The query clause will be cached. But something likes ServiceID, I think it is not suitable.
Please recheck. :)

Reply:
I am wondering, what is the meaning of cache? The data is updated every several seconds, even for the metrics. What is the scope of cache? @dmsolr @aderm Do you have any documentation to explain this?  The official one is better.

Reply:
[https://www.elastic.co/guide/en/elasticsearch/reference/7.5/query-dsl-filtered-query.html](url)
[https://www.elastic.co/guide/en/elasticsearch/client/net-api/7.x/writing-queries.html](url)

> The benefit of executing a query in a filter context is that Elasticsearch is able to forgo calculating a relevancy score, as well as cache filters for faster subsequent performance.

Reply:
Thanks for providing this. Still, the document about `as well as cache filters for faster subsequent performance` is not very clear.  My guess, at our implementation, you moved all the conditions into the filter, then this part of optimization should be none(no subsequent), except for aggregation top N query.

Reply:
The above statement, I feel a bit incomplete
refer this.
[https://www.elastic.co/guide/en/elasticsearch/reference/7.x/query-filter-context.html](url)
First,the search result process does not require scoring, which avoids the process of calculation and sorting. Second, regarding the cache mechanism of the filter, under what circumstances will the cache be used? This is still under study. However, it should be sure at present that most of the ES queries in SW should hit yes or no. They belong to the filter scene and do not need to be scored, except of course for the scenes that are sorted.

Reply:
> However, it should be sure at present that most of the ES queries in SW should hit yes or no.

Yes. The score is not important for us. We use ES like normal storage, rather than a search engine.